src/probnum/filtsmooth/statespace/transition.py

```diff
@@ -1,221 +1,355 @@
 
-        Nb: Think of transition as a verb, i.e. this method "transitions" a realization of a random variable.
 
         Parameters
         ----------
-        real :
-            Realization of the random variable.
-        start :
-            Starting point :math:`t`.
-        stop :
-            End point :math:`t + \\Delta t`.
-        step :
-            Intermediate step-size. Optional, default is None.
-        _diffusion :
-            Optional diffusion parameter for this transition. This field is usually used
-            to update diffusions with calibrated versions thereof.
-        _linearise_at :
-            For approximate transitions , for instance ContinuousEKFComponent,
-            this argument overloads the state at which the Jacobian is computed.
+        rv
+            Random variable that describes the current state.
+        t
+            Current time point.
+        dt
+            Increment :math:`\Delta t`. Ignored for discrete-time transitions.
+        compute_gain
+            Flag that indicates whether the expected gain of the forward transition shall be computed. This is important if the forward-pass
+            is computed as part of a forward-backward pass, as it is for instance the case in a Kalman update.
+        _diffusion
+            Special diffusion of the driving stochastic process, which is used internally.
+        _linearise_at
+            Specific point of linearisation for approximate forward passes (think: extended Kalman filtering). Used internally for iterated filtering and smoothing.
 
         Returns
         -------
         RandomVariable
-            Random variable, describing the state at time :math:`t + \\Delta t`
-            based on realization at time :math:`t`.
-        dict
-            Additional information in form of a dictionary,
-            for instance the cross-covariance in the
-            prediction step, access to which is useful in smoothing.
-
-        See Also
-        --------
-        :meth:`transition_rv`
-            Apply transition to a random variable.
+            New state, after applying the forward-pass.
+        Dict
+            Information about the forward pass. Can for instance contain a `gain` key, if `compute_gain` was set to `True` (and if the transition supports this functionality).
         """
         raise NotImplementedError
 
     @abc.abstractmethod
-    def transition_rv(
-        self,
-        rv: "RandomVariable",
-        start: float,
-        stop: Optional[float] = None,
-        step: Optional[float] = None,
-        _diffusion: Optional[float] = 1.0,
-        _linearise_at: Optional[RandomVariable] = None,
-    ) -> (RandomVariable, Dict):
-        """Transition a random variable from time :math:`t` to time
-        :math:`t+\\Delta t`.
+    def forward_realization(
+        self, real, t, dt=None, compute_gain=False, _diffusion=1.0, _linearise_at=None
```

Comment:
I find the naming `real` misleading. Should we name that `realization` instead?

Reply:
is it misleading even right next to "forward_realization"? One could rename it, I am just a bit worried about line length.

```
abc, info = transition.forward_realization(realization=sth, t=sth, ...)
``` 

vs
```
abc, info = transition.forward_realization(real=sth, t=sth, ...)
```
do you think there is a lot to be gained here? If so, I will consider changing it!

Perhaps @nathanaelbosch has an opinion here?

Reply:
In this case I don't mind having `real`, it seems clear enough to me in the context of a function called `forward_realization` :) But Jonathan might have the most unbiased view on this.

Reply:
Generally, I don't want to let line length compromise the readability of the code. But if both of you think that it's ok, I am fine with it :) I just immediately thought of real in terms of "real part". As in complex numbers. But I guess that's just me then

Reply:
Decision: use `realization`